File: robs_sport_club_backend/models/UserStatistics.js
class UserStatistics {
  constructor(id, childId, monthYear, totalTrainingHours = 0, totalMatchHours = 0) {
    this.id = id;
    this.childId = childId;
    this.monthYear = monthYear;
    this.totalTrainingHours = totalTrainingHours;
    this.totalMatchHours = totalMatchHours;
  }

  // Method to create a `UserStatistics` instance from JSON
  static fromJson(json) {
    return new UserStatistics(
      json.id,
      json.childId,
      new Date(json.monthYear),
      json.totalTrainingHours || 0,
      json.totalMatchHours || 0
    );
  }

  // Method to convert a `UserStatistics` instance to JSON
  toJson() {
    return {
      id: this.id,
      childId: this.childId,
      monthYear: this.monthYear.toISOString(),
      totalTrainingHours: this.totalTrainingHours,
      totalMatchHours: this.totalMatchHours,
    };
  }
}

module.exports = UserStatistics;

File: robs_sport_club/lib/models/userstatistics.dart
class UserStatistics {
  final int id;
  final int childId;
  final DateTime monthYear;
  final double totalTrainingHours;
  final double totalMatchHours;

  UserStatistics({
    required this.id,
    required this.childId,
    required this.monthYear,
    required this.totalTrainingHours,
    required this.totalMatchHours,
  });

  // Factory constructor to create a `UserStatistics` instance from JSON
  factory UserStatistics.fromJson(Map<String, dynamic> json) {
    return UserStatistics(
      id: json['id'],
      childId: json['childId'],
      monthYear: DateTime.parse(json['monthYear']),
      totalTrainingHours: json['totalTrainingHours'].toDouble(),
      totalMatchHours: json['totalMatchHours'].toDouble(),
    );
  }

  // Method to convert a `UserStatistics` instance to JSON
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'childId': childId,
      'monthYear': monthYear.toIso8601String(),
      'totalTrainingHours': totalTrainingHours,
      'totalMatchHours': totalMatchHours,
    };
  }
}
